sdk/monitor/opentelemetry-exporter-azuremonitor/opentelemetry/exporter/azuremonitor/export/__init__.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import json
+import logging
+import typing
+from enum import Enum
+from urllib.parse import urlparse
+
+from opentelemetry.sdk.metrics.export import MetricsExportResult
+from opentelemetry.sdk.trace.export import SpanExportResult
+from opentelemetry.sdk.util import ns_to_iso_str
+from opentelemetry.trace import Span, SpanKind
+from opentelemetry.trace.status import StatusCanonicalCode
+
+from azure.core.exceptions import HttpResponseError
+from azure.core.pipeline.policies import ProxyPolicy, RetryPolicy
+from opentelemetry.exporter.azuremonitor import utils
+from opentelemetry.exporter.azuremonitor._generated import AzureMonitorExporter
+from opentelemetry.exporter.azuremonitor._generated.models import TelemetryItem, TrackResponse
+from opentelemetry.exporter.azuremonitor.options import ExporterOptions
+from opentelemetry.exporter.azuremonitor.storage import LocalFileStorage
+
+logger = logging.getLogger(__name__)
+
+
+class ExportResult(Enum):
+    SUCCESS = 0
+    FAILED_RETRYABLE = 1
+    FAILED_NOT_RETRYABLE = 2
+
+
+# pylint: disable=broad-except
+class BaseExporter:
+    """Azure Monitor base exporter for OpenTelemetry.
+
+    Args:
+        options: :doc:`export.options` to allow configuration for the exporter
+    """
+
+    def __init__(self, **options):
+        self._telemetry_processors = []
+        self.options = ExporterOptions(**options)
+        retry_policy = RetryPolicy(timeout=self.options.timeout)
+        proxy_policy = ProxyPolicy(proxies=self.options.proxies)
+        self.client = AzureMonitorExporter(
+            self.options.endpoint, proxy_policy=proxy_policy, retry_policy=retry_policy)
+        self.storage = LocalFileStorage(
```

Comment:
Curious as to whether there is logic in `azure.core` we can leverage for local storage.